test/server/connection_handler_test.cc

```diff
@@ -1192,18 +1197,23 @@ TEST_F(ConnectionHandlerTest, TcpListenerRemoveFilterChainCalledAfterListenerIsR
 
   EXPECT_CALL(dispatcher_, clearDeferredDeleteList());
   EXPECT_CALL(*access_log_, log(_, _, _, _));
-  handler_->removeListeners(listener_tag);
 
+  {
+    // Filter chain removal in the same poll cycle but earlier.
+    handler_->removeListeners(listener_tag);
+  }
   EXPECT_EQ(0UL, handler_->numConnections());
 
   EXPECT_EQ(0UL, TestUtility::findGauge(stats_store_, "downstream_cx_active")->value());
   EXPECT_EQ(0UL, TestUtility::findGauge(stats_store_, "test.downstream_cx_active")->value());
 
   const std::list<const Network::FilterChain*> filter_chains{filter_chain_.get()};
   MockFunction<void()> on_filter_chain_removed;
-  handler_->removeFilterChains(listener_tag, filter_chains,
-                               [&on_filter_chain_removed]() { on_filter_chain_removed.Call(); });
-
+  {
+    // Listener removal in the same poll cycle but later than filer chain removal.
```

Comment:
filer -> filter

Reply:
Thanks!  Fixed 5 occurrence in the repo.